dhcpv6/dhcpv6message.go

```diff
@@ -65,14 +65,42 @@ func GetTime() uint32 {
 	return uint32((now.Nanoseconds() / 1000000000) % 0xffffffff)
 }
 
+// NewSolicit creates a new SOLICIT message without populating the CID field.
+func NewSolicit(modifiers ...Modifier) (DHCPv6, error) {
+	d, err := NewMessage()
+	if err != nil {
+		return nil, err
+	}
+	d.(*DHCPv6Message).SetMessage(SOLICIT)
+	oro := new(OptRequestedOption)
+	oro.SetRequestedOptions([]OptionCode{
+		DNS_RECURSIVE_NAME_SERVER,
+		DOMAIN_SEARCH_LIST,
+	})
+	d.AddOption(oro)
+	d.AddOption(new(OptElapsedTime))
+	// FIXME use real values for IA_NA
+	iaNa := new(OptIANA)
+	iaNa.IaId = [4]byte{0xfa, 0xce, 0xb0, 0x0c}
+	iaNa.T1 = 0xe10
+	iaNa.T2 = 0x1518
+	d.AddOption(iaNa)
+
+	// Apply modifiers
+	for _, mod := range modifiers {
+		d = mod(d)
+	}
+	return d, nil
+}
+
 // NewSolicitForInterface creates a new SOLICIT message with DUID-LLT, using the
 // given network interface's hardware address and current time
 func NewSolicitForInterface(ifname string, modifiers ...Modifier) (DHCPv6, error) {
-	d, err := NewMessage()
+	// We want modifiers to be applied later
+	d, err := NewSolicit()
```

Comment:
you can further simplify this function by computing the DUID before, and applying all the modifiers via NewSolicit instead of the loop at the end:

```
// get the CID for the specified interface
iface, err := net.InterfaceByName(iface)
if err ...

modifiers = append(modifiers, WithClientID(duid))
d, err := NewSolicit(modifiers)
```

Reply:
something we may want to decide here is whether the CID modifier should take precedence on the other modifiers or not. For example, the caller of `NewSolicitForInterface` also passes `WithClientID(...)`. Do we want this to take precedence, or to give precedence to the `WithClientID` passed inside the function?